kernel/syscall.c

```diff
@@ -1633,6 +1633,42 @@ long myst_syscall_unlink(const char* pathname)
     return ret;
 }
 
+long myst_syscall_unlinkat(int dirfd, const char* pathname, int flags)
+{
+    long ret = 0;
+
+    (void)flags;
+
+    if (flags & ~AT_REMOVEDIR)
+        ERAISE(-EINVAL);
+
+    if (!pathname)
+        ERAISE(-ENOENT);
+
+    if (*pathname == '\0')
```

Comment:
I am skeptical that we can't pass in empty or null pathname to unlinkat. Could you find out by writing a test program and run with Linux?

Reply:
I am guess `unlinkat` would remove the folder `dirfd` if `AT_REMOVEDIR` is specified.

Reply:
@Sahakait I found a post regarding empty pathname: https://bugzilla.kernel.org/show_bug.cgi?id=93441. It seems if pathname is an empty string, then flags must add `AT_EMPTY_PATH`. If that's the case, then the following checks are covered by `myst_get_absolute_path_from_dirfd`. You could remove the check here.

https://github.com/deislabs/mystikos/blob/e33dcd868fb0c06c8534f06f89cdc599158c97f5/kernel/syscall.c#L873-L874

https://github.com/deislabs/mystikos/blob/e33dcd868fb0c06c8534f06f89cdc599158c97f5/kernel/syscall.c#L881-L884

Reply:
Thanks Francis. The issue you referred to is still open, meaning the Linux kernel hasn't implemented `AT_EMPTY_PATH` for `unlinkat` yet. I will resolve this comment.